api/CcsSso.Core.ServiceOnboardingScheduler/Jobs/CASOnboardingJob.cs

```diff
@@ -0,0 +1,498 @@
+    {
+      var org = await _dataContext.Organisation.FirstOrDefaultAsync(or => !or.IsDeleted && or.Id == orgId);
+      org.RightToBuy = true;
+      await _dataContext.SaveChangesAsync();
+      _logger.LogInformation($"Right to buy flag updated for Org id:{org.Id}, Name: {org.LegalName}");
+
+    }
+
+    private async Task AddDefaultRole(int orgId, List<string> adminList)
+    {
+      var roleList = new List<string>();
+      var roles = _appSettings.CASDefaultRoles;
+
+      if (roles == null)
+      {
+        roleList.Add("CAT_USER");
+        roleList.Add("ACCESS_CAAAC_CLIENT");
+        roleList.Add("JAGGAER_USER");
+        roleList.Add("JAEGGER_BUYER"); //As per the discussion with Artur - Change from ACCESS_JAGGAER to JAEGGER_BUYER
+      }
+      else
+      {
+        roleList = roles.ToList();
+      }
+
+
+      var defaultRoles = await _dataContext.CcsAccessRole.Where(ar => !ar.IsDeleted && roleList.Contains(ar.CcsAccessRoleNameKey)).ToListAsync();
+
+      List<OrganisationEligibleRole> addedEligibleRoles = new List<OrganisationEligibleRole>();
+
+      var orgRoles = await _dataContext.OrganisationEligibleRole.Where(x => x.OrganisationId == orgId).ToListAsync();
+
+      defaultRoles.ForEach(async (defaultRole) =>
+      {
+        var alreadyExist = orgRoles.FirstOrDefault(x => x.OrganisationId == orgId && x.CcsAccessRoleId == defaultRole.Id);
+        if (alreadyExist == null)
+        {
+          addedEligibleRoles.Add(new OrganisationEligibleRole
+          {
+            OrganisationId = orgId,
+            CcsAccessRoleId = defaultRole.Id
+          });
+
+          _logger.LogInformation($"Added org role:{defaultRole.CcsAccessRoleNameKey}");
+        }
+        else
+        {
+          _logger.LogInformation($"Org role {defaultRole.CcsAccessRoleNameKey} already exists. org Id  {orgId}");
+        }
+      });
+
+      if (addedEligibleRoles.Count > 0)
+      {
+        _dataContext.OrganisationEligibleRole.AddRange(addedEligibleRoles);
+        await _dataContext.SaveChangesAsync();
+      }
+
+
+
+      foreach (var userName in adminList)
+      {
+        var user = await _dataContext.User
+                  .Include(u => u.Party).ThenInclude(p => p.Person).ThenInclude(p => p.Organisation)
+                  .Include(u => u.UserAccessRoles)
+                  .FirstOrDefaultAsync(u => !u.IsDeleted && u.UserName == userName);
+
+        if (user == null)
+        {
+          _logger.LogInformation($"No user found in user table for user {userName}");
+          continue;
+        }
+
+        List<UserAccessRole> userAccessRoles = new List<UserAccessRole>();
+
+        foreach (var role in defaultRoles)
+        {
+          var organisationAdminAccessRole = await _dataContext.OrganisationEligibleRole
+           .FirstOrDefaultAsync(oer => !oer.IsDeleted && oer.OrganisationId == user.Party.Person.OrganisationId && oer.CcsAccessRole.CcsAccessRoleNameKey == role.CcsAccessRoleNameKey && oer.OrganisationId == orgId);
```

Comment:
You are already getting org id as param no need to take using users i think. Also you are already making call to OrganisationEligibleRole in above code. we can reduce db call.